src/map/status.cpp

```diff
@@ -10501,7 +10507,7 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 			val3 = 5*val1; // SP cost
 			break;
 		case SC_BLESSING:
-			if ((!undead_flag && status->race!=RC_DEMON) || bl->type == BL_PC)
+			if (bl->type == BL_PC || (!undead_flag && status->race != RC_DEMON))
 				val2 = val1;
 			else
 				val2 = 0; // 0 -> Half stat.
```

Comment:
Note: Currently the stats are halved and then rounded down, but actually the decrease should be rounded down.

Basically:
stat -= stat/2;

15 dex should be become 8 dex.